app/lib/submission_builder/document.rb

```diff
@@ -63,5 +63,50 @@ def add_non_zero_claimed_value(xml, elem_name, claimed)
         xml.send(elem_name, claimed: claimed_value)
       end
     end
+
+    def process_mailing_street(xml)
+      return unless @submission.data_source.direct_file_data.mailing_street.present?
+
+      mailing_street = @submission.data_source.direct_file_data.mailing_street
+
+      if mailing_street.length > 30
+        process_long_mailing_street(xml, mailing_street)
+      else
+        xml.MAIL_LN_1_ADR @submission.data_source.direct_file_data.mailing_apartment if @submission.data_source.direct_file_data.mailing_apartment.present?
+        xml.MAIL_LN_2_ADR mailing_street
+      end
+    end
+
+    def process_long_mailing_street(xml, mailing_street)
+      key_found = ["bldg", "bsmt", "dept", "fl", "frnt", "hngr", "key", "lbby", "lot", "lowr", "ofc", "ph", "pier", "rear", "rm", "side", "slip", "spc", "ste", "suite", "stop", "trlr", "unit", "uppr", "Bldg", "Bsmt", "Dept", "Fl", "Frnt", "Hngr", "Key", "Lbby", "Lot", "Lowr", "Ofc", "Ph", "Pier", "Rear", "Rm", "Side", "Slip", "Spc", "Ste", "Suite", "Stop", "Trlr", "Unit", "Uppr","APT", "BLDG", "BSMT", "DEPT", "FL", "FRNT", "HNGR", "KEY", "LBBY", "LOT", "LOWR", "OFC", "PH", "PIER", "REAR", "RM", "SIDE", "SLIP", "SPC", "STE", "SUITE", "STOP", "TRLR", "UNIT", "UPPR"].any? do |key|
```

Comment:
Awesome, looks a lot better @tahsinaislam! Would be great to also do one extra step to move the array into a constant for this file

Something like:

```ruby
KEYS_OR_OTHER_DESCRIPTIVE_NAME = ["bldg", "bsmt", "dept", "fl", "frnt", "hngr", "key", "lbby", "lot", "lowr", "ofc", "ph", "pier", "rear", "rm", "side", "slip", "spc", "ste", "suite", "stop", "trlr", "unit", "uppr", "Bldg", "Bsmt", "Dept", "Fl", "Frnt", "Hngr", "Key", "Lbby", "Lot", "Lowr", "Ofc", "Ph", "Pier", "Rear", "Rm", "Side", "Slip", "Spc", "Ste", "Suite", "Stop", "Trlr", "Unit", "Uppr", "APT", "BLDG", "BSMT", "DEPT", "FL", "FRNT", "HNGR", "KEY", "LBBY", "LOT", "LOWR", "OFC", "PH", "PIER", "REAR", "RM", "SIDE", "SLIP", "SPC", "STE", "SUITE", "STOP", "TRLR", "UNIT", "UPPR"].freeze
``` 

and then reusing this constant in the two places.  This would provide a few benefits:

1. Makes the code more readable/organized
2. Makes the code DRY
3. The frozen literal has performance gains as we are making the object imutable